job-server/src/main/scala/spark/jobserver/util/ManagerLauncher.scala

```diff
@@ -35,76 +85,99 @@ class ManagerLauncher(systemConfig: Config, contextConfig: Config, masterAddress
-      if (deployMode == "client") {
-        val gcFilePath = new File(contextDir, gcFileName).toString()
-        loggingOpts += s" -DLOG_DIR=$contextDir"
-        gcOPTS += s" -Xloggc:$gcFilePath"
-      } else {
-        gcOPTS += s" -Xloggc:$gcFileName"
-      }
+    val gcFileName = getEnvironmentVariable("GC_OUT_FILE_NAME", "gc.out")
+    if (deployMode == "client") {
+      val gcFilePath = new File(contextDir, gcFileName).toString()
+      loggingOpts += s" -DLOG_DIR=$contextDir"
+      gcOPTS += s" -Xloggc:$gcFilePath"
+    } else {
+      gcOPTS += s" -Xloggc:$gcFileName"
+    }
 
-      val contextSparkMaster = Try(contextConfig.getString("launcher.spark.master"))
-        .getOrElse(SparkMasterProvider.fromConfig(systemConfig).getSparkMaster(systemConfig))
-      launcher.setMaster(contextSparkMaster)
-      launcher.setMainClass("spark.jobserver.JobManager")
-      launcher.addAppArgs(masterAddress, contextActorName, getEnvironmentVariable("MANAGER_CONF_FILE"))
-      launcher.addSparkArg("--conf", s"spark.executor.extraJavaOptions=$loggingOpts")
-      launcher.addSparkArg("--driver-java-options",
-          s"$gcOPTS $baseJavaOPTS $loggingOpts $configOverloads $extraJavaOPTS")
-
-      if (!contextName.isEmpty) {
-        launcher.setAppName(contextName)
-      }
+    val contextSparkMaster = Try(contextConfig.getString("launcher.spark.master"))
+      .getOrElse(SparkMasterProvider.fromConfig(systemConfig).getSparkMaster(systemConfig))
+    launcher.setMaster(contextSparkMaster)
+    launcher.setMainClass("spark.jobserver.JobManager")
+    launcher.addAppArgs(masterAddress, contextActorName, getEnvironmentVariable("MANAGER_CONF_FILE"))
+    launcher.addSparkArg("--conf", s"spark.executor.extraJavaOptions=$loggingOpts")
+    launcher.addSparkArg("--driver-java-options",
+      s"$gcOPTS $baseJavaOPTS $loggingOpts $configOverloads $extraJavaOPTS")
+
+    if (!contextName.isEmpty) {
+      launcher.setAppName(contextName)
+    }
 
-      if (useSuperviseMode) {
-        launcher.addSparkArg("--supervise")
-      }
+    if (useSuperviseMode) {
+      launcher.addSparkArg("--supervise")
+    }
 
-      if (contextConfig.hasPath(SparkJobUtils.SPARK_PROXY_USER_PARAM)) {
-         launcher.addSparkArg("--proxy-user", contextConfig.getString(SparkJobUtils.SPARK_PROXY_USER_PARAM))
-      }
+    if (contextConfig.hasPath(SparkJobUtils.SPARK_PROXY_USER_PARAM)) {
+      launcher.addSparkArg("--proxy-user", contextConfig.getString(SparkJobUtils.SPARK_PROXY_USER_PARAM))
+    }
 
-      for (e <- Try(contextConfig.getConfig("launcher"))) {
-         e.entrySet().asScala.map { c =>
-           // Supervise mode was already handled above, no need to do it here again
-           if (c.getKey != "spark.driver.supervise") {
-            launcher.addSparkArg("--conf", s"${c.getKey}=${c.getValue.unwrapped.toString}")
-           }
-         }
+    getStringSeq(contextConfig, "jars").foreach(launcher.addJar)
```

Comment:
Are `jars` and `files` new parameters to context? I do not recall any such parameters previously